core/compilePatterns.ml

```diff
@@ -13,42 +13,59 @@ open Utility
 open Ir
 open SugarConstructors.DummyPositions
 
-type pattern = [
-| `Any
-| `Nil
-| `Cons     of pattern * pattern
-| `Variant  of name * pattern
-| `Effect   of name * pattern list * pattern
-| `Negative of StringSet.t
-| `Record   of (pattern StringMap.t) * pattern option
-| `Constant of Constant.t
-| `Variable of binder
-| `As       of binder * pattern
-| `HasType  of pattern * Types.datatype
-]
+module type CONSTSET = Set with type elt = Constant.t
+module ConstSet = Set.Make(Constant)
+module ConstMap = Map.Make(Constant)
+
+module Pattern =
```

Comment:
I tend to write:

```
module Patter = struct
  type t = ...
```
and I think this is formatting used in many places in our code.